sympy/solvers/ode/systems.py

```diff
@@ -322,8 +326,11 @@ def _first_order_type5_6_subs(A, t, b=None):
     if factor_terms is not None:
         t_ = Symbol("{}_".format(t))
         F_t = integrate(factor_terms[0], t)
-        inverse = solveset(Eq(t_, F_t), t)
 
+        if _is_lambert(F_t,x):
+            inverse = ConditionSet(t_, Eq(F_t,0),S.Complexes)
+        else:
+            inverse = solveset(Eq(t_, F_t), t)
```

Comment:
The `is_lambert` function should not be used here in the ode module.

We need solveset to return the correct result for this case. It is fine if `solveset` returns a `ConditionSet` in some cases.

Reply:
I've tried that part but I've got no clue to execute it. I think it's not possible since an equation when solveset is solving it has no path to return condition set. If we make solveset to return condition set then other solutions will get hindered!